lua/lean/html.lua

```diff
@@ -612,7 +612,7 @@ function BufDiv:buf_make_event_context()
   return {
     rerender = function() self:buf_render() end,
     rehover = function() self:buf_hover() end,
-    root = self,
+    self = self,
```

Comment:
BTW, this is no longer type-correct.  Also, the `clear_all` handler uses the `root` field.

Reply:
Whoops, my bad, fixed for Lean 4. What do you mean by it not being type-correct?

Reply:
You've already fixed it.  The type declaration for `DivEventContext` had a `root` field which wasn't provided here.

(Type checking support in the sumneko language server is still a work in progress and not enabled by default yet afaict.)

Reply:
It's also somewhat annoying to wire up in CI which I gave a quick look at. Didn't see an easy emmy type checker to use though.

Reply:
The only other emmy typechecker I know of is https://github.com/Benjamin-Dobell/IntelliJ-Luanalysis/, which would be even harder to run in CI.  We can probably contribute a PR for a command-line linter mode if we want to integrate the sumneko checker in CI.